src/scvi/model/base/_training_mixin.py

```diff
@@ -84,27 +84,14 @@ def train(
         **kwargs
            Additional keyword arguments passed into :class:`~scvi.train.Trainer`.
         """
-        if datamodule is not None and not self._module_init_on_train:
-            raise ValueError(
-                "Cannot pass in `datamodule` if the model was initialized with `adata`."
-            )
-        elif datamodule is None and self._module_init_on_train:
-            raise ValueError(
-                "If the model was not initialized with `adata`, a `datamodule` must be passed in."
-            )
-
         if max_epochs is None:
-            if datamodule is None:
+            if self.adata is not None:
```

Comment:
We should take here n_obs from summary stats to make it compatible with a dataloader.

Reply:
See below we don't need the if statement.